src/main/java/com/mojang/brigadier/builder/LiteralArgumentBuilder.java

```diff
@@ -9,6 +9,8 @@
 public class LiteralArgumentBuilder<S> extends ArgumentBuilder<S, LiteralArgumentBuilder<S>> {
     private final String literal;
 
+    private boolean isCaseInsensitive = false;
+
     protected LiteralArgumentBuilder(final String literal) {
```

Comment:
Add the additional `ignoreCase` parameter to the constructor.

Then create the method `public static <S> LiteralArgumentBuilder<S> literal(final String name, final boolean ignoreCase) {`

Then modify `literal(String)` to call the created method with `false` so that old code is still compatible.